lld/ELF/InputFiles.cpp

```diff
@@ -655,6 +739,14 @@ template <class ELFT> void ObjFile<ELFT>::parse(bool ignoreComdats) {
     }
   }
 
+  if (hasAArch64BuildAttributes) {
+    // Handle AArch64 Build Attributes and GNU properties:
+    // - Err on mismatched values.
+    // - Store missing values as GNU properties.
+    handleAArch64BAAndGnuProperties<ELFT>(this, ctx, hasGNUProperties,
```

Comment:
If hasGNUProperties is true then we know that ObjFile has already had `andFeatures` and `aarch64PauthAbiCoreInfo` set.

I think this makes `gnuProperty` and its type `GnuPropertiesInfo` redundant. We are already passing `this` to handleAArch64BAAndGnuProperties so we can just read the values from there.